tests/unit-core/test-arraybuffer.c

```diff
@@ -287,6 +287,7 @@ main (void)
   /* Test ArrayBuffer external memory map/unmap */
   {
     const uint32_t buffer_size = 20;
+    /* cppcheck-suppress variableScope */
```

Comment:
False warning, buffer_p can't be moved to the next block, because it is used from JS code: "var array = new Uint8Array(input_buffer);". Of course cppcheck shouldn't know that jerry_create_arraybuffer_external  and register_js_value propagates buffer_p to a JS code. I can't see any better fix than suppressing the warning.

Reply:
Well, what I'm not sure of in the first place is why is there a block for the 3 statements? It's not the body of an if/for/while/... . And by my very subjective opinion, having such "pointless" blocks is ugly. Getting rid of the superfluous curly braces could both improve code quality and make cppcheck happy.

(Unfortunately, tests have quite some of such blocks, it doesn't occur here alone. At least that used to be the case.)

Reply:
I discussed with the author of the code. This block is added to ensure input_buffer isn't used later and VLA is used instead of malloc to test jerry_create_arraybuffer_external without free callback passed.